### PR TITLE
Document exceptions thrown through the API.

### PR DESCRIPTION
**Pull request by patrickvacek**
This list is not exhaustive, but should give a high-level view of what to expect.

### CODE REVIEW

File: include/libaktualizr/aktualizr.h [hunk 1]
```diff
@@ -34,18 +41,40 @@ class Aktualizr {
    * Initialize aktualizr. Any Secondaries should be added before making this
    * call. This will provision with the server if required. This must be called
    * before using any other aktualizr functions except AddSecondary.
+   *
+   * @throw Initializer::Error and subclasses
+   * @throw Uptane::Exception and subclasses
```
Comment (kbushgit) [inline include/libaktualizr/aktualizr.h:46]: Looking into ```Aktualizr/SotaUptaneClient``` classes I can't find the place where the ```Uptane::Exception``` is not handled by the ```SotaUptaneClient```. Is it necessary right now to expose ```Uptane::Exception```, until it is not necessary?

Comment (patrickvacek) [reply include/libaktualizr/aktualizr.h:46]: `SotaUptaneClient::finalizeAfterReboot()` -> `director_repo.dropTargets()`. This is an annoying hole that I'd like to patch up, actually. I too was surprised by this.

**Comment by xaviergully** [comment]
LGTM

File: include/libaktualizr/aktualizr.h (hunk 1 above)
Comment (kbushgit) [reply include/libaktualizr/aktualizr.h:46]: Sorry, could you point me where actually we can get ```Uptane::Exception``` in the ```DirectoRepo::dropTargets``` ,  instead I see that this call can throw ```SQLException```, If I missed nothing...

Comment (patrickvacek) [reply include/libaktualizr/aktualizr.h:46]: `DirectorRepository::dropTargets()` -> `DirectorRepository::resetMeta()` -> `RepositoryCommon::resetRoot()` -> `Root::Root()` -> `Uptane::Root::UnpackSignedObject()`. There's already one Uptane exception there and more in `MetaWithKeys::UnpackSignedObject()`.

This is admittedly something of a stretch, but it's theoretically possible.

**Pull request merged**